src/field_type.cpp

```diff
@@ -124,18 +150,32 @@ void field_type::load( JsonObject &jo, const std::string & )
                   fallback_intensity_level.translucency );
         optional( jao, was_loaded, "convection_temperature_mod", intensity_level.convection_temperature_mod,
                   fallback_intensity_level.convection_temperature_mod );
-        optional( jao, was_loaded, "effect_id", intensity_level.field_effect.id,
-                  fallback_intensity_level.field_effect.id );
-        optional( jao, was_loaded, "effect_min_duration", intensity_level.field_effect.min_duration,
-                  fallback_intensity_level.field_effect.min_duration );
-        optional( jao, was_loaded, "effect_max_duration", intensity_level.field_effect.max_duration,
-                  fallback_intensity_level.field_effect.max_duration );
-        optional( jao, was_loaded, "effect_intensity", intensity_level.field_effect.intensity,
-                  fallback_intensity_level.field_effect.intensity );
-        optional( jao, was_loaded, "effect_body_part", intensity_level.field_effect.bp,
-                  fallback_intensity_level.field_effect.bp );
-        optional( jao, was_loaded, "inside_immune", intensity_level.field_effect.inside_immune,
-                  fallback_intensity_level.field_effect.inside_immune );
+        if( jao.has_array( "effects" ) ) {
+            JsonArray jae = jao.get_array( "effects" );
+            for( size_t j = 0; j < jae.size(); ++j ) {
+                JsonObject joe = jae.next_object();
+                field_effect fe;
+                mandatory( joe, was_loaded, "effect_id", fe.id );
+                optional( joe, was_loaded, "min_duration", fe.min_duration );
+                optional( joe, was_loaded, "max_duration", fe.max_duration );
+                optional( joe, was_loaded, "intensity", fe.intensity );
+                optional( joe, was_loaded, "body_part", fe.bp );
+                optional( joe, was_loaded, "is_environmental", fe.is_environmental );
+                optional( joe, was_loaded, "inside_immune", fe.inside_immune );
+                optional( joe, was_loaded, "outside_immune", fe.outside_immune );
+                optional( joe, was_loaded, "inside_chance", fe.inside_chance );
+                optional( joe, was_loaded, "outside_chance", fe.outside_chance );
+                optional( joe, was_loaded, "field_intensity_delta", fe.field_intensity_delta );
+                fe.env_message = no_translation( jo.get_string( "env_message", "" ) );
+                fe.env_message_npc = no_translation( jo.get_string( "env_message_npc", "" ) );
```

Comment:
Why `no_translation`, shouldn't it be translatable?

Reply:
The translation is done on the fly using `get_env_message`/`get_env_message_npc` functions.

That's per https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/TRANSLATING.md:

Sometimes you may also want to store a string that needs no translation

```c++
const translation text = no_translation( "This string will not be translated" );
```

The string can then be translated/retrieved with the following code

```c++
const std::string translated = text.translated();
```